definition-src/Game/LambdaHack/Core/Random.hs

```diff
@@ -70,11 +85,11 @@ rollFreq fr g = case runFrequency fr of
                                `showFailure` (nameFrequency fr, n, x)
   [(_, x)] -> (x, g)  -- speedup
   fs -> let sumf = foldl' (\ !acc (!n, _) -> acc + n) 0 fs
-            (r, ng) = R.randomR (1, sumf) g
+            (r, ng) = randomR0 (pred sumf) g
             frec :: Int -> [(Int, a)] -> a
             frec !m [] = error $ "impossible roll"
                                  `showFailure` (nameFrequency fr, fs, m)
-            frec m ((n, x) : _) | m <= n = x
+            frec m ((n, x) : _) | m < n = x
```

Comment:
Nicely simplified! I guess dice rolls can also be expressed with `randomR0`.  I will actually try to get rid of the `random` package altogether (except as a dep of splitmix).

Reply:
Hmm, never mind your `randomR` actually does the translation to `randomR0` and doesn't use `random`. :)